### PR TITLE
`IntroEligibility.init(: Int)`: refactored to avoid hardcoding values

### PR DESCRIPTION
**Pull request by NachoSoto**

### CODE REVIEW

**Comment by NachoSoto** [comment]
Types FTW :)

File: Purchases/Public/IntroEligibility.swift [hunk 1]
```diff
@@ -54,17 +56,16 @@ private extension IntroEligibilityStatus {
...
+        guard let result = Self.mapping[statusCode] else {
             throw IntroEligibilityStatusError.invalidStatusCode(statusCode)
         }
+
+        self = result
     }
+
+    private static let mapping: [Int: IntroEligibilityStatus] = Dictionary(
+        uniqueKeysWithValues: IntroEligibilityStatus.allCases.map { ($0.rawValue, $0) }
+    )
```
Comment (aboedo) [inline Purchases/Public/IntroEligibility.swift:68]: would using `RawRepresentable` remove the need to do this mapping ourselves?

**Review by aboedo — APPROVED** [review]
awesome! 🎉

**Review by taquitos — APPROVED** [review]
I like this.

File: Purchases/Public/IntroEligibility.swift (hunk 1 above)
Comment (NachoSoto) [reply Purchases/Public/IntroEligibility.swift:68]: Having an associated `Int` implicitly makes the type `RawRepresentable`. But that doesn't provide this _reverse_ mapping from `rawValue` to the type.

**Pull request merged**